[0.10.0-beta2] panic when collectd enabled

I deleted the database after the upgrade and used the new config format.  Only change to config was to enable collectd. When collectd is disabled it starts up normally.  Full log attached.

OS: RHEL7

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x5c1a2c]

goroutine 85 [running]:
github.com/influxdb/influxdb/services/collectd.(*Service).UnmarshalCollectd(0xc2080625a0, 0xc2096cfb00, 0x0, 0x0, 0x0)
        /tmp/tmp.zI7RBxEcb8/src/github.com/influxdb/influxdb/services/collectd/service.go:302 +0x9cc
github.com/influxdb/influxdb/services/collectd.(*Service).handleMessage(0xc2080625a0, 0xc209318000, 0x51b, 0x5ac)
        /tmp/tmp.zI7RBxEcb8/src/github.com/influxdb/influxdb/services/collectd/service.go:231 +0x28a
github.com/influxdb/influxdb/services/collectd.(*Service).serve(0xc2080625a0)
        /tmp/tmp.zI7RBxEcb8/src/github.com/influxdb/influxdb/services/collectd/service.go:218 +0x2e0
created by github.com/influxdb/influxdb/services/collectd.(*Service).Open
        /tmp/tmp.zI7RBxEcb8/src/github.com/influxdb/influxdb/services/collectd/service.go:141 +0x103d
```

```
[collectd]
  enabled = true
  # bind-address = ""
  database = "collectd"
  typesdb = "/usr/share/collectd/types.db"

  # These next lines control how batching works. You should have this enabled
  # otherwise you could get dropped metrics or poor performance. Batching
  # will buffer points in memory if you have many coming in.

  batch-size = 1000 # will flush if this many points get buffered
  batch-pending = 5 # number of batches that may be pending in memory
  batch-timeout = "1s" # will flush at least this often even if we haven't hit buffer limit
  read-buffer = 0 # UDP Read buffer size, 0 means OS default. UDP listener will fail if set above OS max.
```

[influxdb.txt](https://github.com/influxdata/influxdb/files/104421/influxdb.txt)

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -31,6 +31,7 @@
 - [#5382](https://github.com/influxdata/influxdb/pull/5382): Fixes some escaping bugs with tag keys and values.
 - [#5349](https://github.com/influxdata/influxdb/issues/5349): Validate metadata blob for 'influxd backup'
 - [#5469](https://github.com/influxdata/influxdb/issues/5469): Conversion from bz1 to tsm doesn't work as described
+- [#5449](https://github.com/influxdata/influxdb/issues/5449): panic when dropping collectd points
 
 ## v0.9.6 [2015-12-09]
 

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -32,6 +32,7 @@
 - [#5349](https://github.com/influxdata/influxdb/issues/5349): Validate metadata blob for 'influxd backup'
 - [#5469](https://github.com/influxdata/influxdb/issues/5469): Conversion from bz1 to tsm doesn't work as described
 - [#5449](https://github.com/influxdata/influxdb/issues/5449): panic when dropping collectd points
+- [#5455](https://github.com/influxdata/influxdb/issues/5455): panic: runtime error: slice bounds out of range when loading corrupted wal segment
 
 ## v0.9.6 [2015-12-09]
 

--- a/services/collectd/service.go
+++ b/services/collectd/service.go
@@ -299,7 +299,7 @@ func (s *Service) UnmarshalCollectd(packet *gollectd.Packet) []models.Point {
 		p, err := models.NewPoint(name, tags, fields, timestamp)
 		// Drop invalid points
 		if err != nil {
-			s.Logger.Printf("Dropping point %v: %v", p.Name, err)
+			s.Logger.Printf("Dropping point %v: %v", name, err)
 			s.statMap.Add(statDroppedPointsInvalid, 1)
 			continue
 		}

--- a/tsdb/engine/tsm1/wal.go
+++ b/tsdb/engine/tsm1/wal.go
@@ -493,6 +493,10 @@ func (w *WriteWALEntry) UnmarshalBinary(b []byte) error {
 				}
 			case stringEntryType:
 				length := int(btou32(b[i : i+4]))
+				if i+length > int(uint32(len(b))) {
+					return fmt.Errorf("corrupted write wall entry")
+				}
+
 				i += 4
 				v := string(b[i : i+length])
 				i += length

